atc/worker/image/image_resource_fetcher.go

```diff
@@ -305,8 +323,11 @@ func (i *imageResourceFetcher) getLatestVersion(
 		return nil, err
 	}
 
-	checkingResource := i.resourceFactory.NewResourceForContainer(imageContainer)
-	versions, err := checkingResource.Check(context.TODO(), i.imageResource.Source, nil)
+	processSpec := runtime.ProcessSpec{
+		Path: "/opt/resource/check",
+	}
+	checkingResource := i.resourceFactory.NewResource(i.imageResource.Source, nil, i.imageResource.Version)
+	versions, err := checkingResource.Check(context.TODO(), processSpec, imageContainer)
```

Comment:
Valid to make it a real ctx.
However, this has been using ctx.TODO since prior to the refactor - https://github.com/concourse/concourse/blame/master/atc/worker/image/image_resource_fetcher.go#L309